CombatTagPlus/src/main/java/net/minelink/ctplus/task/TagUpdateTask.java

```diff
@@ -51,6 +51,9 @@ public void run() {
             if (!plugin.getSettings().getUntagMessage().isEmpty()) {
                 player.sendMessage(plugin.getSettings().getUntagMessage());
             }
+
+            plugin.getTagManager().untag(player.getUniqueId());
```

Comment:
@cbarber I think it is best if we guarantee that 'PlayerUntagEvent' is _guaranteed_ to be fired before isUntagged() returns true.
In your implementation, a external plugin could act as if a player is untagged, before the untag event happens.
By only firing the untag event once you call the untag method, you require client code to call the untag method whenever a player gets untagged (as a result of the getTag method returning null), which is error prone, and reveals details of the tagging implementation.
By firing the untag event in a getTag method, you hide the details of the tagging/untagging implementation from the client code, and guarantee that the event is fired before client code realizes that the player is untagged.
While I agree that normally getters shouldn't have side effects, this side effect is transparent to the client, since it makes no difference to them where the event is fired, as long as untag event is fired before anyone is aware of the untagging.